tests/infrastructure/golden_images/update_boot_source/test_boot_sources_vm.py

```diff
@@ -0,0 +1,75 @@
+import logging
+
+import pytest
+from ocp_resources.data_source import DataSource
+from ocp_resources.persistent_volume_claim import PersistentVolumeClaim
+
+from utilities.infra import assert_os_version_mismatch_in_vm, validate_os_info_vmi_vs_linux_os
+from utilities.storage import data_volume_template_with_source_ref_dict
+from utilities.virt import VirtualMachineForTests, running_vm
+
+LOGGER = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def boot_source_preference_from_data_source_dict(data_import_cron_matrix__function__):
+    return data_import_cron_matrix__function__[[*data_import_cron_matrix__function__][0]]["preference"]
+
+
+@pytest.fixture()
+def data_source_from_data_import_cron(
+    golden_images_namespace,
+    golden_images_persistent_volume_claims_scope_function,
+    golden_images_volume_snapshot_scope_function,
+    data_import_cron_matrix__function__,
+):
+    data_source = DataSource(name=[*data_import_cron_matrix__function__][0], namespace=golden_images_namespace.name)
+    if data_source.exists is None:
+        pytest.fail(f"Data Source {[*data_import_cron_matrix__function__][0]} is missing")
+    if data_source.source.kind == PersistentVolumeClaim.kind:
+        cluster_volumes = golden_images_persistent_volume_claims_scope_function
+    else:
+        cluster_volumes = golden_images_volume_snapshot_scope_function
+    assert any([data_source.name in volume.name for volume in cluster_volumes]), (
+        f"DataSource data_source {data_source.source.kind} {data_source.name} is missing"
+    )
+    return data_source
+
+
+@pytest.fixture()
+def auto_update_boot_source_instance_type_vm(
+    unprivileged_client,
+    namespace,
+    modern_cpu_for_migration,
+    data_source_from_data_import_cron,
+):
+    LOGGER.info(f"Create a VM using {data_source_from_data_import_cron.name} dataSource")
+    with VirtualMachineForTests(
+        client=unprivileged_client,
+        name=f"{data_source_from_data_import_cron.name}-data-source-vm",
+        namespace=namespace.name,
+        vm_instance_type_infer=True,
+        vm_preference_infer=True,
+        data_volume_template=data_volume_template_with_source_ref_dict(
+            data_source=data_source_from_data_import_cron,
+        ),
+        cpu_model=modern_cpu_for_migration,
+    ) as vm:
+        running_vm(vm=vm)
+        yield vm
+
+
+@pytest.mark.polarion("CNV-7586")
+def test_instance_type_vm_from_auto_update_boot_source(
+    latest_fedora_release_version,
+    auto_update_boot_source_instance_type_vm,
+    boot_source_preference_from_data_source_dict,
+):
+    LOGGER.info(f"Verify {auto_update_boot_source_instance_type_vm.name} OS version and virtctl info")
+    if "fedora" in auto_update_boot_source_instance_type_vm.name and latest_fedora_release_version:
+        boot_source_preference_from_data_source_dict = f"fedora{latest_fedora_release_version}"
```

Comment:
Why do you need this? And why do you need this for fedora only?

Reply:
If DataSource is Ready, and DataImportCron is UpToDate - then we'll have the latest image

Reply:
The data source link might point us to a different image.
Infra team is responsible for the DataSource images (data import cron), so we should validate they point to the latest version

Reply:
Maybe image validation should be a separate test? 
Also, what does `latest_fedora_release_version` returns? We are not using its return value anywhere here

Reply:
This test purpose is to validate the OS inside, there is no meaning if we don't verify it. do you think we should rename the test? I'm ok with that if you think it does not represent the test purpose, and also open for any name suggestions.

`latest_fedora_release_version` return a string, for example "40" or "41 , we use it to create a string for `expected_os`